src/collectors/ping/test/testping.py

```diff
@@ -48,18 +48,44 @@ def test_should_work_with_real_data_bad_gentoo(self, publish_mock):
     @patch('os.access', Mock(return_value=True))
     @patch.object(Collector, 'publish')
     def test_should_work_with_real_data_host_gentoo(self, publish_mock):
-        patch_communicate = patch(
-            'subprocess.Popen.communicate',
-            Mock(return_value=(
-                self.getFixture('host_gentoo').getvalue(),
-                '')))
+        config = get_collector_config('PingCollector', {
+            'interval': 10,
+            'target_a': 'localhost',
+            'target_b': 'www.example.org',
+            'target_c': '192.168.0.1',
+            'bin': 'true'
+        })
 
-        patch_communicate.start()
-        self.collector.collect()
-        patch_communicate.stop()
+        collector = PingCollector(config, None)
+
+        def popen_se(command, **kwargs):
```

Comment:
It is needed because the order in which the calls for various hosts are made is arbitrary; we cannot know which host will be pinged first. This results from the loop at https://github.com/python-diamond/Diamond/blob/master/src/collectors/ping/ping.py#L60, combined with the implementation details of `.keys()` (see https://docs.python.org/2/library/stdtypes.html#dict.items).